### PR TITLE
Updating `python-py` to 1.10.0 to fix CVE-2020-29651

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Fixing CVE-2020-29651 by updating `python-py` to 1.10.0. According to the authors the library is in maintenance mode and no significant changes have been made since 1.6.0.

###### Change Log  

- Fixed CVE-2020-29651.

###### Does this affect the toolchain?  

No.

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2020-29651

###### Test Methodology

- Local package build for `python-py` and `pytest`.

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -4955,8 +4955,8 @@
         "type": "other",
         "other": {
           "name": "python-py",
-          "version": "1.6.0",
-          "downloadUrl": "https://github.com/pytest-dev/py/archive/1.6.0.tar.gz"
```
Comment (mrgirgin) [inline cgmanifest.json:4959]: Shouldn't we update this line to use PyPI as well?

Comment (PawelWMS) [reply cgmanifest.json:4959]: You're probably right, since that's what we're building with.

**Review by thcrain-msft — APPROVED** [review]
Looks good to me! 

Small suggestion for the future: PyPI actually has alternate URLs that are easier to read and easier to upgrade from version to version. They're of the form:

```
https://pypi.io/packages/source/<first letter of PyPI package name>/<PyPI package name>/<source tarball filename>
```

In this instance, it would be `https://pypi.io/packages/source/p/py/py-1.10.0.tar.gz`.

Don't bother changing it now since it's very minor and you've already tested the build, just something to consider in the future.

**Comment by PawelWMS** [comment]
> Looks good to me!
> 
> Small suggestion for the future: PyPI actually has alternate URLs that are easier to read and easier to upgrade from version to version. They're of the form:
> 
> ```
> https://pypi.io/packages/source/<first letter of PyPI package name>/<PyPI package name>/<source tarball filename>
> ```
> 
> In this instance, it would be `https://pypi.io/packages/source/p/py/py-1.10.0.tar.gz`.
> 
> Don't bother changing it now since it's very minor and you've already tested the build, just something to consider in the future.

Neat! Since I'm lazy I just go to PyPI and copy the link from their web page, but I'll keep that in mind, thanks!

**Pull request merged**